src/pip/_internal/utils/misc.py

```diff
@@ -923,3 +925,14 @@ def pairwise(iterable):
     """
     iterable = iter(iterable)
     return zip_longest(iterable, iterable)
+
+
+def partition(
+    pred,  # type: Callable[[T], bool]
+    iterable,  # type: Iterable[T]
+):
+    # type: (...) -> Tuple[Iterable[T], Iterable[T]]
+    'Use a predicate to partition entries into false entries and true entries'
```

Comment:
nitpick: we tend to write docstring with triple double quotes ?

Reply:
I'll fix this. To explain the inconsistency, this was copied directly from [itertools-recipes](https://docs.python.org/3/library/itertools.html#itertools-recipes).

Reply:
Fixed in d241f0d3.